Assets/MixedRealityToolkit-SDK/Features/Audio/Influencers/IAudioInfluencer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.Audio.Influencers
```

Comment:
Shouldn't this be in core interfaces?

Reply:
no

Reply:
Why not? Seems like something that should be. What if a developer whats to make their own influencers?

What's the advantage of using the interface if we're not going to let people use their own concrete implementations?

Reply:
they CAN add their own concrete implementatioons…. in the SDK like AudioOccluder is a concrete implementation

this is one of our "use it if you like" components

Reply:
Then what's the apprehension to adding it to core?

Reply:
because it has never been a core feature. it is not necessary to 99% of apps

Reply:
I would argue that audio is a pretty core feature.

Reply:
This is advanced audio. Most apps will be fine with Unity's built in audio support and not need to add these features on top of it.

Reply:
Indeed, but in that case, shouldn't we be registering it as a `IMixedRealityComponent` so we know we're adding it to the correct platforms?

I don't see us checking SDK's or plugins in this PR, but we do set it, but what if we're not targeting WMR platform?

Reply:
Which scripts in this PR depend on a specific platform or plugin?

Reply:
Good question.

Reply:
The spatializer is being set in the Unity project for the example scene. This PR does not handle that, but is instead a port of audio influencers and effects.

For setting the spatializer, see #2592.

Reply:
Okay, well what about devs who import the MRTK into their project? Will they have to manually set it as well? Do we have a way to warn them if it's unset? or if their platform is unsupported?

Reply:
How do we handle other spatializers for other platforms? would this demo still apply?

Reply:
Again, for most of that, see #2592.

This PR uses stock Unity components. If the resulting audio ends up spatialized or not, it doesn't really matter. These simply add effects on top of the audio.

Reply:
Because they might not have the SDK. Remember, the SDK is only an interpretation of the interfaces, which are all part of the framework.

Reply:
If they don't have the SDK, there's nothing to talk to the interfaces. Why do I need the interfaces if the script that consumes them doesn't exist?

I'm aware of what the SDK is.

Reply:
I don't think you guys understand the point I'm trying to make. 

Either we use an interface and we move it to core, or we keep a concrete type and remove the interface.

Reply:
I understand that that's what you're saying.

I'm saying that that's not a requirement. It's fine as-is.

Reply:
It's a requirement per the architectural agreement we laid out before we started.

Reply:
Can you point me to that requirement? The interface layer mentions devices, interactions, and input, none of which this seems to fall under.

Reply:
It is not a catch-all layer, nor should it be, as the size of the core should be kept as minimal as possible.

Reply:
https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/MRTK-vNext.md#objective

Reply:
> It is not a catch-all layer, nor should it be, as the size of the core should be kept as minimal as possible.

Agreed, which is why it's only and interface defining the structure, not the concrete implementation.

Reply:
Essentially this class belongs in the "interface layer"

Reply:
fine. i'll move the interface. i still do not agree that EVERY interface belongs in the core, but i dont want to argue anymore about an SDK component port.

Reply:
there are likely to be more PRs where there are sdk layer specific interfaces...

Reply:
I just want to make it clear that we're not just dragging and dropping code from the htk.

We need to be thoughtful of the way we've structured things, and make sure we're sticking to what we agreed on with the initial proposed architecture.